### PR TITLE
Adding advance example and documentation

### PR DESCRIPTION
**Pull request by sanujkul**
Adding an advance example that can make use of software serial as old Arduino boards only have one Serial and thus using your library with **SoftwareSerial** output Stream will give lots of **flexibility.**  

Adding **documentation** as it will help the users to know how to use this library and what all ways are there.

### CODE REVIEW

**Comment by sanujkul** [comment]
Can someone please help me in understanding why I have failed this check and what has to be done?
I have only added an example and edited Readme.md.

**Comment by per1234** [comment]
>Can someone please help me in understanding why I have failed this check

The check is for a specific formatting standard, which your code doesn't match.

>what has to be done?

In this case, you can resolve the problem by doing a **Tools > Auto Format** on `/examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino` if using the Arduino IDE or pressing **Ctrl + b** if using the Arduino Web Editor.

I know it's confusing that the Arduino IDE automatically adds these `*` to multi-line comments, then the auto format takes them away (https://github.com/arduino/Arduino/issues/3941), but we used the auto format style in our formatting standard.

File: examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Advanced Debug can be helpful in embedded applications when 
+ * there are more that two microcontrollers connected serially
+ * or a wireless sensor like Xbee is connected to the serial port 
+ * that will send data wirelessly to other Xbee node.
+ * 
+ * In boards like Ardunino Nano, UNO, MEGA only one serial port is available,
+ * therefore additional Software Serial ports can be made usinf SoftwareSerial
```
Comment (per1234) [inline examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino]:
```suggestion
 * therefore additional Software Serial ports can be made using SoftwareSerial
```

File: examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Advanced Debug can be helpful in embedded applications when 
+ * there are more that two microcontrollers connected serially
+ * or a wireless sensor like Xbee is connected to the serial port 
+ * that will send data wirelessly to other Xbee node.
+ * 
+ * In boards like Ardunino Nano, UNO, MEGA only one serial port is available,
```
Comment (per1234) [inline examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino]:
```suggestion
 * In boards like Arduino Nano, UNO, MEGA only one serial port is available,
```

File: examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Advanced Debug can be helpful in embedded applications when 
+ * there are more that two microcontrollers connected serially
+ * or a wireless sensor like Xbee is connected to the serial port 
+ * that will send data wirelessly to other Xbee node.
```
Comment (per1234) [inline examples/Arduino_Debug_Advance/Arduino_Debug_Advance.ino]:
```suggestion
 * or a wireless sensor like XBee is connected to the serial port 
 * that will send data wirelessly to other XBee node.
```

File: README.md [hunk 4]
```diff
@@ -27,3 +27,37 @@ If desired timestamps can be prefixed to the debug message. Timestamp output can
...
 Normally all debug output is redirected to the primary serial output of each board (`Serial`). In case you want to redirect the output to another output stream you can make use of `setDebugOutputStream(&Serial2)`.
+
+# Documentation
+### 1. Debug :
+Arduino_DebugUtils Object that will be used for calling member functions.
+
+### 2. Debug.setDebugLevel(int const debug_level) :
+Parameter debug_level in order of lowest to highest priority are : `DBG_NONE`, `DBG_ERROR`, `DBG_WARNING`, `DBG_INFO` (default), `DBG_DEBUG`, and `DBG_VERBOSE`. 
+Return type: void.
+Example:
```
Comment (per1234) [inline README.md]:
```suggestion

Return type: void.

Example:
```
Unfortunately, Markdown doesn't render single line breaks. This caused the original rendered output to look like:
>... , and `DBG_VERBOSE`. Return type: void. Example:

There are several ways to cause a line break in the rendered output, but I think the extra vertical whitespace caused by the empty lines I added makes things more readable.

File: README.md [hunk 5]
```diff
@@ -27,3 +27,37 @@ If desired timestamps can be prefixed to the debug message. Timestamp output can
...
+```
+Debug.setDebugLevel(DBG_VERBOSE);
+```
+### 2. Debug.setDebugOutputStream(Stream * stream) :
+By default, Output Stream is Serial. In advanced cases other objects could be other serial ports (if available), or can be a Software Serial object.
+Example:
```
Comment (per1234) [inline README.md]:
```suggestion

Example:
```
See explanation in the comment above.

**Comment by sanujkul** [comment]
@per1234 . Thank you! I have committed your suggestions and also modified Readme.md to remove numbering and added Return type and Example to all member functions. But still it is failing travis test!

**Comment by sanujkul** [comment]
@per1234 . Thank you very much for your suggestions.  I did auto-formatting and it worked! :)

**Review by aentinger — APPROVED** [review]
LGTM :+1: Thank you @sanujkul :wave:

**Pull request merged**